### PR TITLE
dts: Add Huawei Mediapad T1 A21L

### PR DESCRIPTION
**Pull request by Lagertonne**
Proof, if necessary: 
photo5303291550152897048

### CODE REVIEW

File: dts/msm8916/msm8916-huawei-hwt1a21l.dts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+/include/ "skeleton.dtsi"
+
+/ {
+	qcom,msm-id = <206 0>;
+	qcom,board-id = <8065 0> , <8065 4>;
```
Comment (Minecrell) [inline dts/msm8916/msm8916-huawei-hwt1a21l.dts:9]: Please include only the board-id you have, probably:

```suggestion
	qcom,board-id = <8065 4>;
```

Also, can you post the `lk_log` or UART (see README) so I can check if everything looks okay?

**Comment by Lagertonne** [comment]
Below is the content of `output.txt` created by `fastboot oem lk_log && fastboot get_staged output.txt`

```
[0] smem ram ptable found: ver: 1 len: 4
[0] Minor socinfo format detected: 0.8
[0] smem ram ptable found: ver: 1 len: 4
[0] SCM call: 0x2000601 failed with :fffffffc
[0] Failed to initialize SCM
[0] welcome to lk

[0] calling constructors
[0] initializing heap
[0] initializing threads
[0] initializing dpc
[0] initializing timers
[0] creating bootstrap completion thread
[10] top of bootstrap2()
[10] initializing platform
[10] platform_init()
[10] initializing target
[10] target_init()
[20]  Initializing MMC host data structure and clock!
[40] Decoded CID fields:
[50] Manufacturer ID: 90
[50] OEM ID: 0x14a
[50] Product Name: HAG2e
[50] Product revision: 0.7
[50] Product serial number: 16F16089
[60] Manufacturing date: 3 2004
[70] Decoded CSD fields:
[70] cmmc_structure: 3
[80] card_cmd_class: f5
[80] write_blk_len: 512
[80] read_blk_len: 512
[80] r2w_factor: 2
[80] sector_size: 0
[90] c_size_mult:7
[90] c_size: 4095
[90] nsac_clk_cycle: 100
[90] taac_ns: 15000000
[90] tran_speed: 0 kbps
[100] erase_blk_len: 1
[100] read_blk_misalign: 0
[100] write_blk_misalign: 0
[100] read_blk_partial: 0
[110] write_blk_partial: 0
[110] wp_grp_size: 16384
[110] Card Capacity: 15762194432 Bytes
[110] SDHC Running in HS200 mode
[130] Done initialization of the card
[130] erase_grpsize: 31
[130] erase_grpmult: 31
[140] wp_grpsize: 15
[140] wp_grpen: 1
[140] perm_wp: 0
[140] temp_wp: 0
[150] pm8x41_get_is_cold_boot: cold boot
[150] calling apps_init()
[150] Board: platform: 206, foundry: 0x0, platform_version: 0x10000, platform_hw: 0x1f81, platform_subtype: 0x4, target: 0x4010081, baseband: 0x0, platform_hlos_subtype: 0x0
[170] pmic_info[0]: type: 0x1000b, version: 0x20000, target: 0x2000b
[170] Command line from primary bootloader: lk2nd androidboot.emmc=true androidboot.serialno=16f16089 androidboot.huawei_swtype=normal androidboot.huawei_bootmode=boot androidboot.localproppath=vdf/de usb.serial=C5R6R16804000521 androidboot.baseband=msm mdss_mdp.panel=1:dsi:0:qcom,mdss_dsi_boe_nt51017_10_800p_video huawei,lcd_panel_id = 0
[200] Device model: Huawei Mediapad T1 A21L
[210] Display Init: Start
[210] Continuous splash detected: pipe: 0, base: 0x83200000, stride: 2400, source: 800x1280 (0,0), output: 800x1280 (0,0) (cmd mode: 0)
[220] smem ram ptable found: ver: 1 len: 4
[220] smem ram ptable found: ver: 1 len: 4
[230] Display Init: Done
[230] serial number: 16f16089
[230] ptn[0]:Name[sbl1] Size[1024] Type[0] First[34] Last[1057]
[240] ptn[1]:Name[hyp] Size[1024] Type[0] First[1058] Last[2081]
[240] ptn[2]:Name[sec] Size[64] Type[0] First[8192] Last[8255]
[250] ptn[3]:Name[DDR] Size[64] Type[0] First[8256] Last[8319]
[260] ptn[4]:Name[rpm] Size[1024] Type[0] First[16384] Last[17407]
[260] ptn[5]:Name[tz] Size[1024] Type[0] First[17408] Last[18431]
[270] ptn[6]:Name[aboot] Size[10112] Type[0] First[18432] Last[28543]
[270] ptn[7]:Name[pad] Size[2048] Type[0] First[28544] Last[30591]
[280] ptn[8]:Name[oeminfo] Size[131072] Type[0] First[30592] Last[161663]
[290] ptn[9]:Name[modemst1] Size[8192] Type[0] First[161664] Last[169855]
[290] ptn[10]:Name[modemst2] Size[8192] Type[0] First[169856] Last[178047]
[300] ptn[11]:Name[modem] Size[196608] Type[0] First[180224] Last[376831]
[310] ptn[12]:Name[fsg] Size[8192] Type[0] First[376832] Last[385023]
[310] ptn[13]:Name[fsc] Size[4096] Type[0] First[385024] Last[389119]
[320] ptn[14]:Name[ssd] Size[4096] Type[0] First[389120] Last[393215]
[320] ptn[15]:Name[log] Size[131072] Type[0] First[393216] Last[524287]
[330] ptn[16]:Name[persist] Size[65536] Type[0] First[524288] Last[589823]
[340] ptn[17]:Name[boot] Size[39936] Type[0] First[590848] Last[630783]
[340] ptn[18]:Name[recovery] Size[49152] Type[0] First[630784] Last[679935]
[350] ptn[19]:Name[misc] Size[8192] Type[0] First[679936] Last[688127]
[360] ptn[20]:Name[cache] Size[524288] Type[0] First[688128] Last[1212415]
[360] ptn[21]:Name[system] Size[3670016] Type[0] First[1212416] Last[4882431]
[370] ptn[22]:Name[cust] Size[1572864] Type[0] First[4882432] Last[6455295]
[380] ptn[23]:Name[userdata] Size[24330207] Type[0] First[6455296] Last[30785502]
[380] ptn[24]:Name[lk2nd] Size[1024] Type[0] First[589824] Last[590847]
[390] fastboot_init()
[390] serial number: 16f16089
[490] USB init ept @ 0x8f66b000
[510] udc_start()
[540] creating fastboot menu keys detect thread
[700] -- reset --
[700] -- portchange --
[850] -- reset --
[860] -- portchange --
[1050] fastboot: processing commands
[13670] fastboot: oem lk_log
```

I attached the UART output of a boot as a file: [uart_output.txt](https://github.com/msm8916-mainline/lk2nd/files/6023562/uart_output.txt)

**Review by Minecrell — APPROVED** [review]

**Comment by hicode002** [comment]
> Below is the content of `output.txt` created by `fastboot oem lk_log && fastboot get_staged output.txt`
> 
> ```
> [0] smem ram ptable found: ver: 1 len: 4
> [0] Minor socinfo format detected: 0.8
> [0] smem ram ptable found: ver: 1 len: 4
> [0] SCM call: 0x2000601 failed with :fffffffc
> [0] Failed to initialize SCM
> [0] welcome to lk
> 
> [0] calling constructors
> [0] initializing heap
> [0] initializing threads
> [0] initializing dpc
> [0] initializing timers
> [0] creating bootstrap completion thread
> [10] top of bootstrap2()
> [10] initializing platform
> [10] platform_init()
> [10] initializing target
> [10] target_init()
> [20]  Initializing MMC host data structure and clock!
> [40] Decoded CID fields:
> [50] Manufacturer ID: 90
> [50] OEM ID: 0x14a
> [50] Product Name: HAG2e
> [50] Product revision: 0.7
> [50] Product serial number: 16F16089
> [60] Manufacturing date: 3 2004
> [70] Decoded CSD fields:
> [70] cmmc_structure: 3
> [80] card_cmd_class: f5
> [80] write_blk_len: 512
> [80] read_blk_len: 512
> [80] r2w_factor: 2
> [80] sector_size: 0
> [90] c_size_mult:7
> [90] c_size: 4095
> [90] nsac_clk_cycle: 100
> [90] taac_ns: 15000000
> [90] tran_speed: 0 kbps
> [100] erase_blk_len: 1
> [100] read_blk_misalign: 0
> [100] write_blk_misalign: 0
> [100] read_blk_partial: 0
> [110] write_blk_partial: 0
> [110] wp_grp_size: 16384
> [110] Card Capacity: 15762194432 Bytes
> [110] SDHC Running in HS200 mode
> [130] Done initialization of the card
> [130] erase_grpsize: 31
> [130] erase_grpmult: 31
> [140] wp_grpsize: 15
> [140] wp_grpen: 1
> [140] perm_wp: 0
> [140] temp_wp: 0
> [150] pm8x41_get_is_cold_boot: cold boot
> [150] calling apps_init()
> [150] Board: platform: 206, foundry: 0x0, platform_version: 0x10000, platform_hw: 0x1f81, platform_subtype: 0x4, target: 0x4010081, baseband: 0x0, platform_hlos_subtype: 0x0
> [170] pmic_info[0]: type: 0x1000b, version: 0x20000, target: 0x2000b
> [170] Command line from primary bootloader: lk2nd androidboot.emmc=true androidboot.serialno=16f16089 androidboot.huawei_swtype=normal androidboot.huawei_bootmode=boot androidboot.localproppath=vdf/de usb.serial=C5R6R16804000521 androidboot.baseband=msm mdss_mdp.panel=1:dsi:0:qcom,mdss_dsi_boe_nt51017_10_800p_video huawei,lcd_panel_id = 0
> [200] Device model: Huawei Mediapad T1 A21L
> [210] Display Init: Start
> [210] Continuous splash detected: pipe: 0, base: 0x83200000, stride: 2400, source: 800x1280 (0,0), output: 800x1280 (0,0) (cmd mode: 0)
> [220] smem ram ptable found: ver: 1 len: 4
> [220] smem ram ptable found: ver: 1 len: 4
> [230] Display Init: Done
> [230] serial number: 16f16089
> [230] ptn[0]:Name[sbl1] Size[1024] Type[0] First[34] Last[1057]
> [240] ptn[1]:Name[hyp] Size[1024] Type[0] First[1058] Last[2081]
> [240] ptn[2]:Name[sec] Size[64] Type[0] First[8192] Last[8255]
> [250] ptn[3]:Name[DDR] Size[64] Type[0] First[8256] Last[8319]
> [260] ptn[4]:Name[rpm] Size[1024] Type[0] First[16384] Last[17407]
> [260] ptn[5]:Name[tz] Size[1024] Type[0] First[17408] Last[18431]
> [270] ptn[6]:Name[aboot] Size[10112] Type[0] First[18432] Last[28543]
> [270] ptn[7]:Name[pad] Size[2048] Type[0] First[28544] Last[30591]
> [280] ptn[8]:Name[oeminfo] Size[131072] Type[0] First[30592] Last[161663]
> [290] ptn[9]:Name[modemst1] Size[8192] Type[0] First[161664] Last[169855]
> [290] ptn[10]:Name[modemst2] Size[8192] Type[0] First[169856] Last[178047]
> [300] ptn[11]:Name[modem] Size[196608] Type[0] First[180224] Last[376831]
> [310] ptn[12]:Name[fsg] Size[8192] Type[0] First[376832] Last[385023]
> [310] ptn[13]:Name[fsc] Size[4096] Type[0] First[385024] Last[389119]
> [320] ptn[14]:Name[ssd] Size[4096] Type[0] First[389120] Last[393215]
> [320] ptn[15]:Name[log] Size[131072] Type[0] First[393216] Last[524287]
> [330] ptn[16]:Name[persist] Size[65536] Type[0] First[524288] Last[589823]
> [340] ptn[17]:Name[boot] Size[39936] Type[0] First[590848] Last[630783]
> [340] ptn[18]:Name[recovery] Size[49152] Type[0] First[630784] Last[679935]
> [350] ptn[19]:Name[misc] Size[8192] Type[0] First[679936] Last[688127]
> [360] ptn[20]:Name[cache] Size[524288] Type[0] First[688128] Last[1212415]
> [360] ptn[21]:Name[system] Size[3670016] Type[0] First[1212416] Last[4882431]
> [370] ptn[22]:Name[cust] Size[1572864] Type[0] First[4882432] Last[6455295]
> [380] ptn[23]:Name[userdata] Size[24330207] Type[0] First[6455296] Last[30785502]
> [380] ptn[24]:Name[lk2nd] Size[1024] Type[0] First[589824] Last[590847]
> [390] fastboot_init()
> [390] serial number: 16f16089
> [490] USB init ept @ 0x8f66b000
> [510] udc_start()
> [540] creating fastboot menu keys detect thread
> [700] -- reset --
> [700] -- portchange --
> [850] -- reset --
> [860] -- portchange --
> [1050] fastboot: processing commands
> [13670] fastboot: oem lk_log
> ```
> 
> I attached the UART output of a boot as a file: [uart_output.txt](https://github.com/msm8916-mainline/lk2nd/files/6023562/uart_output.txt)

Tell me, where is the uart point on this board?
thanks!

**Comment by Lagertonne** [comment]
Sure, I just added a description for how to access the UART in the PMOS-Wiki: https://wiki.postmarketos.org/wiki/Huawei_Mediapad_T1_10_A21L_(hwt1a21l)

tl;dr: Remove the backplate, the four pads next to the screen-cable are relevant for the UART (GND, TX, RX, 3.3V iirc)

**Comment by hicode002** [comment]
> Sure, I just added a description for how to access the UART in the PMOS-Wiki: https://wiki.postmarketos.org/wiki/Huawei_Mediapad_T1_10_A21L_(hwt1a21l)
> 
> tl;dr: Remove the backplate, the four pads next to the screen-cable are relevant for the UART (GND, TX, RX, 3.3V iirc)

thanks

**Comment by hicode002** [comment]
> Sure, I just added a description for how to access the UART in the PMOS-Wiki: https://wiki.postmarketos.org/wiki/Huawei_Mediapad_T1_10_A21L_(hwt1a21l)
> 
> tl;dr: Remove the backplate, the four pads next to the screen-cable are relevant for the UART (GND, TX, RX, 3.3V iirc)

Can you send me a stock kernel uart log?I need to make comparions to make mainline display work.

**Comment by Lagertonne** [comment]
Sadly I don't have any. But I'm very eager to see your fixes for the display, since I couldn't get it working.

You can also reach me on Matrix (@lagertonne:leitstelle511.net) or IRC (lagertonne @hackint), then we don't clog up the issue here :)

**Comment by hicode002** [comment]
> Sadly I don't have any. But I'm very eager to see your fixes for the display, since I couldn't get it working.
> 
> You can also reach me on Matrix (@Lagertonne:leitstelle511.net) or IRC (lagertonne @hackint), then we don't clog up the issue here :)

Now I successfully make the  display work!Now audio,sensors,display,led,vib,wifi,bluetooth,touchscreen ,all of them work！
My matrix id @hicode_002
